src/ast/passes/semantic_analyser.cpp

```diff
@@ -1747,6 +1753,22 @@ If you're seeing errors, try clamping the string sizes. For example:
                         << arg.type().GetTy() << " provided)";
       }
     }
+  } else if (call.func == "socket_cookie") {
+    const auto &type = call.vargs.at(0).type();
+    if (!type.IsPtrTy() || !type.GetPointeeTy() ||
+        !type.GetPointeeTy()->IsRecordTy()) {
+      call.addError() << call.func
```

Comment:
nit: you can just make a small lambda at the top that sets the error so you don't have to repeat this string